test/functional/gridfs.test.js

```diff
@@ -3884,4 +3884,37 @@ describe('GridFS', function() {
       });
     }
   });
+
+  /**
+   * @ignore
+   */
+  it('should correctly create index', {
+    metadata: {
+      requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
+    },
+
+    // The actual test we wish to run
+    test: function(done) {
+      var configuration = this.configuration;
+      var GridStore = configuration.require.GridStore;
+      var client = configuration.newClient(configuration.writeConcernMax(), { poolSize: 1 });
+
+      client.connect(function(err, client) {
+        expect(err).to.not.exist;
+        var db = client.db(configuration.db);
+        var gridStore = new GridStore(db, 'test_gs_save_empty_file', 'w');
+        gridStore.open(function(err) {
+          expect(err).to.not.exist;
+          db.collection('fs.files').createIndex({ filename: 1, uploadDate: 1 }, {}, (err, val) => {
```

Comment:
you can use `this.defer(() => gridStore.close())` here, and remove the `close` call later. The same with `client` after connect. Then you can just call `done` below.